apiserver/facades/client/application/application.go

```diff
@@ -401,6 +416,34 @@ func (api *APIv6) Deploy(args params.ApplicationsDeployV6) (params.ErrorResults,
 	return api.APIBase.Deploy(newArgs)
 }
 
+// Deploy fetches the charms from the charm store and deploys them
+// using the specified placement directives.
+// V12 deploy did not CharmOrigin, so pass through an unknown source.
+func (api *APIv12) Deploy(args params.ApplicationsDeployV12) (params.ErrorResults, error) {
+	var newArgs params.ApplicationsDeploy
+	for _, value := range args.Applications {
+		newArgs.Applications = append(newArgs.Applications, params.ApplicationDeploy{
+			ApplicationName:  value.ApplicationName,
+			Series:           value.Series,
+			CharmURL:         value.CharmURL,
+			CharmOrigin:      &params.CharmOrigin{Source: "unknown"},
```

Comment:
For older facade versions, the charm origin source will be charm-store or local.  Can we derive the value from the CharmURL?

Reply:
Unknown should be inferred as an error once we wire up charm url